lib/script.js

```diff
@@ -182,7 +182,12 @@ function wrapAsyncFunctions(asyncFunctions, sandbox, events, done, sandboxRoot)
             }
           };
         } else {
-          args.push(function() {
+          // CONVENTION:
+          // functions that use promises start with the name 'promisedXXXX', add a callback counter for them
+          if ((/^promised/).test(asyncFunctions[k].name)) {
+            events.emit('addCallback');
+          }
+          args.push(function () {
             if (sandboxRoot._error) return;
             events.emit('finishCallback');
```

Comment:
We hook onto the promise to call this: https://github.com/deployd/deployd/pull/484/files#diff-7e23141f9dd7c0c72d781aa717bc1a7eR90 when the promise is finished, which guarantees that the callback count is decremented.